### PR TITLE
Update Amazon S3 connector docs with latest release

### PR DESCRIPTION
**Pull request by SanojPunchihewa**
## Purpose
- With https://github.com/wso2-extensions/esb-connector-amazons3/pull/54 when the MI/EI server is running in an EC2 instance, the AWS SDK will use the IAM role for authentication when credentials are not defined.
- With https://github.com/wso2-extensions/esb-connector-amazons3/pull/56 a new parameter is introduced to set the content of the file via payload.
- Fixes https://github.com/wso2/docs-apim/issues/4704

## Goals
- Update the `init` operation with steps to set the IAM role for authentication.
- Update `putObject`, `multipartUpload` and `uploadPart` operations docs with the newly added `fileContent` parameter.
- Update the S3 connector example to use the message payload as the file content.

## User stories
- Updated `putObject` operation with the newly added `fileContent` parameter.
s3-connector-put-object-doc

- Updated S3 connector example.
amazons3-connector-synapse-config
amazons3-connector-testing

### CODE REVIEW

File: en/docs/reference/connectors/amazons3-connector/amazons3-connector-example.md [hunk 1]
```diff
@@ -204,47 +183,62 @@ We can use Curl or Postman to try the API. The testing steps are provided for cu
 
 ### Creating a bucket in Amazon S3
 
-1. Create a file called data.xml with the following content. Note that the bucket region is `us-east-2`. If you need to create the bucket in a different region, modify the hard coded region of the API configuration accordingly.
-    ```xml
-    <createBucket>
-        <bucketName>wso2engineers</bucketName>
-        <bucketRegion>us-east-2</bucketRegion>
-    </createBucket>
+1. Create a file called `data.json` with the following content. Note that the bucket region is `us-east-2`. If you need to create the bucket in a different region, modify the hard coded region of the API configuration accordingly.
```
Comment (samgnaniah) [inline en/docs/reference/connectors/amazons3-connector/amazons3-connector-example.md:186]: The us-east-2 region config isn't there in the json sample below. Can we modify this?

File: en/docs/reference/connectors/amazons3-connector/amazons3-connector-example.md [hunk 2]
```diff
@@ -34,127 +34,110 @@ Follow these steps to set up the Integration Project and import AmazonS3 connect
...
 
-1. Right click on the created Integration Project and select, -> **New** -> **Rest API** to create the REST API.
-   <img src="{{base_path}}/assets/img/integrate/connectors/adding-an-api.jpg" title="Adding a Rest API" width="800" alt="Adding a Rest API"/>
+## Creating the Integration Logic
 
-2. Specify the API name as `S3ConnectorTestAPI` and API context as `/s3connector`. You can go to the source view of the XML configuration file of the API and copy the following configuration.
+1. Specify the API name as `S3ConnectorTestAPI` and API context as `/s3connector`.
+
+2. First we will create the `/createbucket` resource. This API resource will retrieve the bucket name from the incoming HTTP PUT request and create a bucket in Amazon S3.
+Right click on the API Resource and go to **Properties** view. We use a URL template called `/createbucket` as we have three API resources inside a single API. The method will be PUT.
```
Comment (samgnaniah) [inline en/docs/reference/connectors/amazons3-connector/amazons3-connector-example.md:42]: I don't think the numbering will work unless we tab this.

File: en/docs/reference/connectors/amazons3-connector/amazons3-connector-example.md [hunk 3]
```diff
@@ -34,127 +34,110 @@ Follow these steps to set up the Integration Project and import AmazonS3 connect
...
+
+    <img src="{{base_path}}/assets/img/integrate/connectors/s3-connector-2x/s3-connector-create-bucket-api.jpg" title="Adding the createbucket resource" width="800" alt="Amazon S3 use case"/>
+
+3. Next drag and drop the 'createBucket' operation of the S3 Connector to the Design View as shown below. Here we are going to receive the following inputs from the user,
```
Comment (samgnaniah) [inline en/docs/reference/connectors/amazons3-connector/amazons3-connector-example.md:46]:
```suggestion
3. Next drag and drop the 'createBucket' operation of the S3 Connector to the Design View as shown below. Here, you will receive the following inputs from the user.
```

File: en/docs/reference/connectors/amazons3-connector/amazons3-connector-example.md [hunk 4]
```diff
@@ -34,127 +34,110 @@ Follow these steps to set up the Integration Project and import AmazonS3 connect
...
+    <img src="{{base_path}}/assets/img/integrate/connectors/s3-connector-2x/s3-connector-drop-create-bucket.jpg" title="Adding the createbucket operation" width="800" alt="Amazon S3 use case"/>
+
+4. Create a connection from the properties window by clicking on the '+' icon as shown below.
+
+    <img src="{{base_path}}/assets/img/integrate/connectors/s3-connector-2x/s3-create-bucket-connection-configs.jpg" title="Creating a new connection" width="800" alt="Amazon S3 use case"/>
+
+    In the popup window, the following parameters must be provided.
+
+    - Connection Name - Unique name to identify the connection by.
+    - Connection Type - Type of the connection which specifies the protocol to be used.
```
Comment (samgnaniah) [inline en/docs/reference/connectors/amazons3-connector/amazons3-connector-example.md:58]:
```suggestion
    - Connection Type - Type of the connection that specifies the protocol to be used.
```

File: en/docs/reference/connectors/amazons3-connector/amazons3-connector-example.md [hunk 5]
```diff
@@ -34,127 +34,110 @@ Follow these steps to set up the Integration Project and import AmazonS3 connect
...
+
+    <img src="{{base_path}}/assets/img/integrate/connectors/s3-connector-2x/s3-create-bucket-connection-configs.jpg" title="Creating a new connection" width="800" alt="Amazon S3 use case"/>
+
+    In the popup window, the following parameters must be provided.
+
+    - Connection Name - Unique name to identify the connection by.
+    - Connection Type - Type of the connection that specifies the protocol to be used.
+    - AWS Access Key ID - Access key associated with your Amazon user account.
+    - AWS Secret Access Key - Secret Access key associated with your Amazon user account.
+    - Region - Region which is used to select a regional endpoint to make requests
```
Comment (samgnaniah) [inline en/docs/reference/connectors/amazons3-connector/amazons3-connector-example.md:61]:
```suggestion
    - Region - Region that is used to select a regional endpoint to make requests
```

File: en/docs/reference/connectors/amazons3-connector/amazons3-connector-example.md [hunk 6]
```diff
@@ -34,127 +34,110 @@ Follow these steps to set up the Integration Project and import AmazonS3 connect
...
+    In the popup window, the following parameters must be provided.
+
+    - Connection Name - Unique name to identify the connection by.
+    - Connection Type - Type of the connection that specifies the protocol to be used.
+    - AWS Access Key ID - Access key associated with your Amazon user account.
+    - AWS Secret Access Key - Secret Access key associated with your Amazon user account.
+    - Region - Region that is used to select a regional endpoint to make requests.
+
+    !!! note
+        1. You can either defince the credentials or let the the AWS SDK to manage the credentials. The SDK will look for AWS credentials in system/user environment variables or use the IAM role for authentication if the application is running in an EC2 instance.
```
Comment (samgnaniah) [inline en/docs/reference/connectors/amazons3-connector/amazons3-connector-example.md:64]:
```suggestion
        1. You can either define the credentials or allow the AWS SDK to manage the credentials. The SDK will look for AWS credentials in system/user environment variables or use the IAM role for authentication if the application is running in an EC2 instance.
```

File: en/docs/reference/connectors/amazons3-connector/amazons3-connector-example.md (hunk 1 above)
Comment (SanojPunchihewa) [reply en/docs/reference/connectors/amazons3-connector/amazons3-connector-example.md:186]: As mentioned, the us-east-2 is hardcoded in the API, therefore we don't need to send it in the JSON payload.

**Review by samgnaniah — APPROVED** [review]

**Comment by samgnaniah** [comment]
Please send the same fix to master branch.

File: en/docs/reference/connectors/amazons3-connector/amazons3-connector-example.md (hunk 2 above)
Comment (SanojPunchihewa) [reply en/docs/reference/connectors/amazons3-connector/amazons3-connector-example.md]: It's a single step, I have mistakenly added a new line. Fixed with a27bfd5

**Pull request merged**